mathesar_ui/src/component-library/select/Select.svelte

```diff
@@ -6,58 +6,57 @@
     ListBoxOptions,
   } from '@mathesar-component-library-dir/list-box';
   import { Dropdown } from '@mathesar-component-library-dir/dropdown';
-  import type { LabelGetter } from '@mathesar-component-library-dir/common/utils/formatUtils';
   import { getLabel as defaultGetLabel } from '@mathesar-component-library-dir/common/utils/formatUtils';
-  import type { Appearance } from '@mathesar-component-library-dir/types';
   import { getGloballyUniqueId } from '@mathesar-component-library-dir/common/utils/domUtils';
   import StringOrComponent from '../string-or-component/StringOrComponent.svelte';
+  import type { SelectProps } from './SelectTypes';
 
   type Option = $$Generic;
+  type $$Props = SelectProps<Option>;
+  type DefinedProps = Required<$$Props>;
 
   const dispatch = createEventDispatcher<{ change: Option | undefined }>();
 
-  export let id = getGloballyUniqueId();
-
-  export let disabled = false;
+  export let id: DefinedProps['id'] = getGloballyUniqueId();
+  export let disabled: DefinedProps['disabled'] = false;
 
   /**
    * Specifies the key on which the options label is stored.
    */
-  export let labelKey = 'label';
+  export let labelKey: DefinedProps['labelKey'] = 'label';
 
-  export let getLabel: LabelGetter<Option | undefined> = (
-    o: Option | undefined,
-  ) => defaultGetLabel(o, labelKey);
+  export let getLabel: DefinedProps['getLabel'] = (o: Option | undefined) =>
+    defaultGetLabel(o, labelKey);
 
   /**
    * List of options to select from. Must be an array of SelectOption.
    * @required
    */
-  export let options: Option[] = [];
+  export let options: DefinedProps['options'];
 
-  export let prependBlank = false;
+  export let prependBlank: DefinedProps['prependBlank'] = false;
 
-  export let value: Option | undefined = undefined;
+  export let value: $$Props['value'] = undefined;
 
   /**
    * Classes to apply to the content (each of the options).
    */
-  export let contentClass = '';
+  export let contentClass: DefinedProps['contentClass'] = '';
 
   /**
    * Classes to apply to the trigger button (the dropdown button).
    */
-  export let triggerClass = '';
+  export let triggerClass: DefinedProps['triggerClass'] = '';
```

Comment:
No additional benefit other than consistency. Since we are explicitly exporting an interface in the component, it makes sense for to mention the types to all the props from that interface.